From 6d0f20bc3d2c663619bdc908db0337f2afee83ec Mon Sep 17 00:00:00 2001
From: chandavath <rcrajagopal@habana.ai>
Date: Mon, 12 Feb 2024 11:21:49 +0200
Subject: [PATCH] [SW-137121] gaudi3: erom pcie: fix compilation issue

Added weak function to pass through the erom pcie build.

Change-Id: I7f75e64b00e60524efcc035c369da8bccc1b4dbd
Signed-off-by: chandavath <rcrajagopal@habana.ai>
---

diff --git a/app/ext_rom/erom_pcie.c b/app/ext_rom/erom_pcie.c
index 8c060c3..3f3108f 100644
--- a/app/ext_rom/erom_pcie.c
+++ b/app/ext_rom/erom_pcie.c
@@ -156,6 +156,7 @@
 
 __weak void ppboot_security_init(void) {}
 __weak void ppboot_secure_arc(void) {}
+__weak bool preboot_security_fuse_override(void) { return false; }
 __weak void erom_efuse_init(void) {}
 
 void erom_main_function(void)
diff --git a/app/ppboot/ppboot.c b/app/ppboot/ppboot.c
index f2864fa..5a8f209 100644
--- a/app/ppboot/ppboot.c
+++ b/app/ppboot/ppboot.c
@@ -264,7 +264,6 @@
 __weak void ppboot_check_stack_mon_dev_specific(void) {}
 __weak void preboot_delay(void) {}
 __weak void sram_set_extension(void) {}
-__weak u32 hl_skip_fw_cfg(void) { return 0; }
 
 void ppboot_init_stack_monitor(void)
 {
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index e2dbccb..b8f5a8f 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1428,6 +1428,8 @@
 	return 0;
 }
 
+__weak u32 hl_skip_fw_cfg(void) { return 0; }
+
 u32 sp_sram_readl(u32 offset)
 {
 	u32 reg = *((volatile u32 *) (((u32) (HL_SP_SRAM_BASE + offset))));
